packages/task/src/browser/task-service.ts

```diff
@@ -0,0 +1,105 @@
+export interface TaskTerminalWidgetFactory {
+    (options: TerminalWidgetOptions): TerminalWidget;
+}
+
+@injectable()
+export class TaskService {
+
+    constructor(
+        @inject(FrontendApplication) protected readonly app: FrontendApplication,
+        @inject(ITaskServer) protected readonly taskServer: ITaskServer,
+        @inject(ILogger) protected readonly logger: ILogger,
+        @inject(WidgetManager) protected readonly widgetManager: WidgetManager,
+        @inject(TaskTerminalWidgetFactory) protected readonly terminalWidgetFactory: TaskTerminalWidgetFactory,
+        @inject(TaskWatcher) protected readonly taskWatcher: TaskWatcher,
+        @inject(MessageService) protected readonly messageService: MessageService,
+        @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService
+    ) {
+        this.taskWatcher.onTaskOutputEntryFound((event: ITaskOutputEntryFoundEvent) => {
+            if (ProblemMarker.is(event.entry)) {
+                // problem marker
+                this.messageService.info(`Task ${event.taskId} has sent a problem marker`);
+                // TODO: handler markers
+
+            } else {
+                // eventually other type of markers
+                logger.error(`Received unsupported type of marker for task ${event.taskId}`);
+            }
+
+        });
+
+        // set workspace root for the backend task server serving this instance
+        this.workspaceService.root.then(root => {
+            this.taskServer.setWorkspaceRoot(root.uri);
+        });
+
+        this.taskWatcher.onTaskExit((event: ITaskExitedEvent) => {
+            if (event.code === 0) {
+                this.messageService.info(`Success: Task ${event.taskId} has completed. Exit code: ${event.code}, signal: ${event.signal}`);
+            } else {
+                this.messageService.error(`Error: Task ${event.taskId} has completed. Exit code: ${event.code}, signal: ${event.signal}`);
+            }
+        });
+
+    }
+
+    public async getTasks(): Promise<string[]> {
+        let tasks: string[];
+
+        tasks = await this.taskServer.getTasks();
+        return tasks;
+    }
+
+    public async run(task: string): Promise<void> {
+        let taskInfo;
+        try {
+            taskInfo = await this.taskServer.run(task);
+        } catch (error) {
+            this.logger.error(`Error launching task '${task}': ${error}`);
+            this.messageService.error(`Error launching task '${task}': ${error}`);
+            return;
+        }
+
+        this.logger.debug(`Task created. task id: ${taskInfo.taskId}, OS ProcessId: ${taskInfo.osProcessId} `);
+
+        // open terminal widget:
+        if (taskInfo.terminalId !== undefined) {
+            this.attachTerminal(taskInfo.terminalId, taskInfo.taskId);
+        }
+    }
+
+    protected attachTerminal(terminalId: number, taskId: number): void {
+        const newTerminal = this.terminalWidgetFactory({
```

Comment:
I wanted to avoid having the task terminal(s) be managed by the widget manager. e.g. I do not want those terminals to be saved/restored by the frontend (make no sense, I think for tasks).

Reply:
I think it should expend the widget manager to allow these features and still use it otherwise we're duplicating code.

Reply:
now using widget manager to create the terminal widget for tasks

Reply:
Opened issue #776, asking about widget restoration be optional